lab_1_keywords_tfidf/main.py

```diff
@@ -17,7 +28,16 @@ def clean_and_tokenize(text: str) -> Optional[list[str]]:
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if isinstance(text, str):
+        punctuation = '''!()-[]{};:'"\,<>./?@#$%^&*_~'''
+        res = " "
+        for element in text:
+            if element not in punctuation:
+                res += element
+        cleaned_text = res.lower().split()
+        return cleaned_text
+    else:
```

Comment:
it is better to change the order of branches here 
```
if <condition>:
    <shorter block of code>
else:
   <longer  block of code>
```
This way code will become more readable

Reply:
This applies to the rest of functions as well

Reply:
So, if I understand you correctly, it should look like this:
    
    if not isinstance(text, str):
        return None
    else:
        punctuation = '''!()-[]{};:'"\,<>./?@#$%^&*_~'''
        res = ""
        for element in text:
            if element not in punctuation:
                res += element
        cleaned_text = res.lower().split()
        return cleaned_text

Reply:
fix this issue in all the other functions as well, please 
in other words, you should try to do an early return, where you firsly check whether data is valid and return None otherwise, and only afterwards implement the main logic 
it is considered more readable